src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -153,13 +158,39 @@ public static boolean isProxySet() {
      *
      * @return String: the set proxy address, null if none are set
      */
-    public static String getProxy() {
+    public static String getProxy() throws InvalidProxyException {
         if (!isProxySet()) {
             return null;
         }
+        String proxy = getProgramProperty(PROXY);
+        String[] proxyParts = proxy.split(":");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "proxy" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)

Reply:
no it can't be, line 162 would have caught it, and thrown the error